memory/src/main/scala/filodb.memory/format/vectors/Histogram.scala

```diff
@@ -369,15 +369,16 @@ object HistogramBuckets {
   }
 
   // NOTE: must point to u16/Short length prefix bytes
-  def apply(bucketsDef: Ptr.U8, formatCode: Byte): HistogramBuckets = formatCode match {
-    case HistFormat_Geometric_Delta  => geometric(UnsafeUtils.ZeroArray, bucketsDef.add(2).addr, false)
-    case HistFormat_Geometric1_Delta => geometric(UnsafeUtils.ZeroArray, bucketsDef.add(2).addr, true)
-    case HistFormat_Custom_Delta     => custom(UnsafeUtils.ZeroArray, bucketsDef.addr)
+  def apply(acc: MemoryReader, bucketsDef: Ptr.U8, formatCode: Byte): HistogramBuckets = formatCode match {
+    case HistFormat_Geometric_Delta  => geometric(acc.base, acc.baseOffset + bucketsDef.add(2).addr, false)
+    case HistFormat_Geometric1_Delta => geometric(acc.base, acc.baseOffset + bucketsDef.add(2).addr, true)
+    case HistFormat_Custom_Delta     => custom(acc.base, bucketsDef.addr)
     case _                           => emptyBuckets
   }
 
   // Create geometric buckets definition
-  def geometric(bucketsDefBase: Array[Byte], bucketsDefOffset: Long, minusOne: Boolean): HistogramBuckets =
+  // FIXME Use MemoryAccessor methods here instead of unsafe access.
+  def geometric(bucketsDefBase: Any, bucketsDefOffset: Long, minusOne: Boolean): HistogramBuckets =
```

Comment:
I'd like to keep `bucketsDefBase` an `Array[Byte]` as that is a bit safer.

Reply:
Needs to be Any because we pass MemoryAccessor.base to this method. Byte Buffers could also be used.

Reply:
Byte buffers do not actually contain memory - they refer to something else.  It would be a problem if byte buffers were passed as a base.